lib/spack/spack/platforms/windows.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import platform
+
+import archspec.cpu
+
+import spack.target
+from spack.operating_systems.windows_os import WindowsOs
+
+from ._platform import Platform
+
+
+class Windows(Platform):
+    priority    = 101
+
+    def __init__(self):
+        super(Windows, self).__init__('windows')
+
+        for name in archspec.cpu.TARGETS:
+            self.add_target(name, spack.target.Target(name))
+
+        self.default = archspec.cpu.host().name
+        self.front_end = self.default
+        self.back_end = self.default
+
+        windows_os = WindowsOs()
+
+        self.default_os = str(windows_os)
+        self.front_os   = str(windows_os)
+        self.back_os    = str(windows_os)
+
+        self.add_operating_system(str(windows_os), windows_os)
+
+    @classmethod
+    def detect(cls):
+        plat = platform.system().lower()
+        return 'cygwin' in plat or 'win32' in plat or 'windows' in plat
```

Comment:
This breaks Cygwin support:
- Cygwin python does not have encoding 'mbcs'
- Attempts to determine the platform version on Cygwin will return the Cygwin version (3.3) not the Windows version (10 or 11)

This will crash when `Windows.detect()` succeeds and the loop tries to call `Windows()`, which calls `WindowsOs()`, which will pass `encoding="mbcs"` to `subprocess.Popen` in some python versions and complain if `platform.release() < 10`